Giving it a second look, the first unit test is failing with:
```
root: ERROR: Error getting block location from MSPileup for /GammaGammaToEE_Elastic_Pt15_8TeV-lpair/Summer12-START53_V7C-v1/GEN-SIM: name 'dataItem' is not defined
Traceback (most recent call last):
  File "/build/cmsbld/jenkins/workspace/DMWM-WMCorePy3-PR-unittests/SLICE/8/label/cms-dmwm-cc7/code/src/python/WMCore/WorkQueue/Policy/Start/StartPolicyInterface.py", line 295, in getDatasetLocationsFromMSPileup
    queryDict = {'query': {'pileupName': dataItem},
NameError: name 'dataItem' is not defined
```

This error is reported in the middle of those 100s of log lines in this unit test: https://cmssdt.cern.ch/dmwm-jenkins/job/DMWM-WMCore-PR-test/14809/testReport/junit/WMCore_t.WorkQueue_t.WorkQueue_t/WorkQueueTest/testPileupOnProduction/

--- a/src/python/WMCore/WorkQueue/Policy/Start/StartPolicyInterface.py
+++ b/src/python/WMCore/WorkQueue/Policy/Start/StartPolicyInterface.py
@@ -12,6 +12,7 @@ from WMCore.WorkQueue.Policy.PolicyInterface import PolicyInterface
 from WMCore.WorkQueue.DataStructs.WorkQueueElement import WorkQueueElement
 from WMCore.DataStructs.LumiList import LumiList
 from WMCore.WorkQueue.WorkQueueExceptions import WorkQueueWMSpecError, WorkQueueNoWorkError
+from WMCore.Services.MSPileup.MSPileupUtils import getPileupDocs
 from dbs.exceptions.dbsClientException import dbsClientException
 from WMCore.Services.CRIC.CRIC import CRIC
 from WMCore.Services.Rucio.Rucio import Rucio
@@ -159,10 +160,10 @@ class StartPolicyInterface(PolicyInterface):
         self.validate()
         try:
             pileupDatasets = self.wmspec.listPileupDatasets()
+            self.logger.debug(f'pileupDatasets: {pileupDatasets}')
             if pileupDatasets:
                 # unwrap {"url":[datasets]} structure into list of datasets
-                datasets = [d for prec in pileupDatasets.values() for d in prec]
-                self.pileupData = self.getDatasetLocations(datasets)
+                self.pileupData = self.getDatasetLocationsFromMSPileup(pileupDatasets)
             self.split()
         # For known exceptions raise custom error that will fail the workflow.
         except dbsClientException as ex:
@@ -271,6 +272,35 @@ class StartPolicyInterface(PolicyInterface):
             locations = self.rucio.getDataLockedAndAvailable(name=datasetPath,
                                                              account=account)
             result[datasetPath] = self.cric.PNNstoPSNs(locations)
+        return result
+    
+    def getDatasetLocationsFromMSPileup(self, datasetsWithDbsURL):
+        """
+        Returns a dictionary with the location of the datasets according to MSPileup
+        :param datasetsWithDbsURL: a dict with the DBS URL as the key, and the associated list of datasets as the value
+        """
+        
+        result = {}
+        for dbsUrl, datasets in datasetsWithDbsURL.items():
+            pileUpinstance = '-testbed' if 'cmsweb-testbed' in dbsUrl else '-prod'
+            msPileupUrl = f'https://cmsweb{pileUpinstance}.cern.ch/ms-pileup/data/pileup'
+            self.logger.info(f'Will fetch {len(datasets)} from MSPileup url: {dbsUrl}')
+            for dataset in datasets:
+                queryDict = {'query': {'pileupName': dataset},
+                            'filters': ['expectedRSEs', 'currentRSEs', 'pileupName', 'containerFraction', 'ruleIds']}
+                try:
+                    doc = getPileupDocs(msPileupUrl, queryDict, method='POST')[0]
+                    currentRSEs = doc['currentRSEs']
+                    self.logger.debug(f'Retrieved MSPileup document: {doc}')
+                    if len(currentRSEs) == 0:
+                        self.logger.warning(f'No RSE has a copy of the desired pileup dataset. Expected RSEs: {doc["expectedRSEs"]}')  
+                    result[dataset] = doc['currentRSEs']
+                except IndexError:
+                    self.logger.warning('Did not find any pileup document for query: %s', queryDict['query'])
+                    result[dataset] = []
+                except Exception as ex:
+                    self.logger.exception('Error getting block location from MSPileup for %s: %s', dataset, str(ex))
+
         return result
 
     def blockLocationRucioPhedex(self, blockName):

--- a/test/python/WMCore_t/WorkQueue_t/Policy_t/Start_t/Block_t.py
+++ b/test/python/WMCore_t/WorkQueue_t/Policy_t/Start_t/Block_t.py
@@ -32,6 +32,25 @@ parentProcArgs["SplittingAlgo"] = "LumiBased"
 parentProcArgs["LumisPerJob"] = 8
 
 
+# A mock class for WMCore.WorkQueue.Policy.Start.Block
+class ThisBlock(Block):
+    def __init__(self, **args):
+        super().__init__(**args)
+
+    def getDatasetLocationsFromMSPileup(self, datasetsWithDbsURL):
+        """
+        Override StartPolicyInterface.getDatasetLocationsFromMSPileup
+        for unit test purposes
+        """
+        print("Mocked ThisBlock.getDatasetLocationsFromMSPileup ")
+        expectedLocations = ['T2_XX_SiteA', 'T2_XX_SiteB', 'T2_XX_SiteC']
+        result = {}
+        for dbsUrl, datasets in datasetsWithDbsURL.items():
+            for dataset in datasets:
+                result[dataset] = expectedLocations
+        return result
+
+
 def rerecoWorkload(workloadName, arguments, assignArgs=None):
     factory = ReRecoWorkloadFactory()
     wmspec = factory.factoryWorkloadConstruction(workloadName, arguments)
@@ -434,7 +453,8 @@ class BlockTestCase(EmulatedUnitTestCase):
         if it is present in the workload.
         """
         for task in MultiTaskProcessingWorkload.taskIterator():
-            units, _, _ = Block(**self.splitArgs)(MultiTaskProcessingWorkload, task)
+            # units, _, _ = Block(**self.splitArgs)(MultiTaskProcessingWorkload, task)
+            units, _, _ = ThisBlock(**self.splitArgs)(MultiTaskProcessingWorkload, task)
             self.assertEqual(58, len(units))
             for unit in units:
                 pileupData = unit["PileupData"]

